src/main/java/com/fasterxml/jackson/core/io/IOContext.java

```diff
@@ -157,46 +137,62 @@ public IOContext(StreamReadConstraints src, StreamWriteConstraints swc, BufferRe
      *
      * @since 2.16
      */
-    public IOContext(StreamReadConstraints src, StreamWriteConstraints swc, BufferRecycler br,
-                     ContentReference contentRef, boolean managedResource, ErrorReportConfiguration erc)
+    public IOContext(StreamReadConstraints src, StreamWriteConstraints swc, ErrorReportConfiguration erc,
+            BufferRecycler br, ContentReference contentRef, boolean managedResource)
     {
         _streamReadConstraints = src;
         _streamWriteConstraints = swc;
+        _errorReportConfiguration = erc;
         _bufferRecycler = br;
         _contentReference = contentRef;
         _sourceRef = contentRef.getRawContent();
         _managedResource = managedResource;
-        _errorReportConfiguration = erc;
     }
 
     /**
+     * Deprecated legacy constructor.
+     *
      * @param src constraints for streaming reads
      * @param br BufferRecycler to use, if any ({@code null} if none)
      * @param contentRef Input source reference for location reporting
      * @param managedResource Whether input source is managed (owned) by Jackson library
      *
      * @since 2.15
-     * @deprecated Since 2.16. Use {@link #IOContext(StreamReadConstraints, StreamWriteConstraints, BufferRecycler, 
-     * ContentReference, boolean, ErrorReportConfiguration)} instead.
+     * @deprecated Since 2.16. Use {@link #IOContext(StreamReadConstraints, StreamWriteConstraints, 
+     * ErrorReportConfiguration, BufferRecycler, ContentReference, boolean)} instead.
      */
     @Deprecated
     public IOContext(StreamReadConstraints src, BufferRecycler br,
                      ContentReference contentRef, boolean managedResource)
     {
-        this(src, StreamWriteConstraints.defaults(), br, contentRef, managedResource, ErrorReportConfiguration.defaults());
+        this(src, StreamWriteConstraints.defaults(), ErrorReportConfiguration.defaults(),
+                br, contentRef, managedResource);
+    }
+
+    @Deprecated // In 2.16: remove ASAP
```

Comment:
Moved from further up; only needed temporarily since it'd have been added in 2.16 so once factories/builders call proper main constructor, this can be safely removed.